gobblin-metrics/src/main/java/gobblin/metrics/MetricContext.java

```diff
@@ -141,7 +144,10 @@ private MetricContext(String name, MetricContext parent, List<Tag<?>> tags,
 
   private synchronized ExecutorService getExecutorService() {
     if(!this.executorServiceOptional.isPresent()) {
-      this.executorServiceOptional = Optional.of(Executors.newCachedThreadPool());
+      this.executorServiceOptional = Optional.of(MoreExecutors.getExitingExecutorService(
```

Comment:
Please use gobblin.util.ExecutorsUtils.newThreadFactory(Optional<Logger>, Optional<String>) or a variation of it as by default uncaught exceptions in thread pools will go completely undetected.